aidial_adapter_vertexai/chat/static_tools.py

```diff
@@ -27,21 +27,62 @@ def parse_gemini_tools(
     ) -> List[GeminiTool] | None: ...
 
 
+class DynamicRetrievalConfig(BaseModel):
+    mode: Literal["MODE_DYNAMIC", "MODE_UNSPECIFIED"] | None = None
+    dynamic_threshold: float | None = None
+
+    @classmethod
+    def root_validator(cls, values):
+        if (
+            values.get("mode") == "MODE_UNSPECIFIED"
+            and values.get("dynamic_threshold") is not None
+        ):
+            raise ValidationError(
+                "dynamic_threshold must be None when mode is MODE_UNSPECIFIED"
+            )
+        if values.get("dynamic_threshold") is not None:
+            threshold = values.get("dynamic_threshold")
+            if threshold < 0 or threshold > 1:
+                raise ValidationError(
+                    "dynamic_threshold must be between 0 and 1"
+                )
+        return values
+
+
+class GoogleSearchConfig(BaseModel):
+    class Config:
+        extra = "forbid"
+
+    dynamic_retrieval_config: DynamicRetrievalConfig | None = None
+
+
 class GoogleSearchGroundingTool(StaticToolProcessor):
     @staticmethod
     def parse_gemini_tools(
         static_function: StaticFunction,
     ) -> List[GeminiTool] | None:
         if static_function.name == ToolName.GOOGLE_SEARCH:
+            google_search_config = GoogleSearchConfig()
             if static_function.configuration:
-                raise ValidationError(
-                    "Google search tool doesn't support configuration"
-                )
-            return [
-                GeminiTool.from_google_search_retrieval(
-                    grounding.GoogleSearchRetrieval()
+                try:
+                    google_search_config = GoogleSearchConfig.validate(
+                        static_function.configuration
+                    )
+                except PydanticValidationError:
+                    raise ValidationError(
+                        "Invalid configuration for Google search tool"
+                    )
+            tools = [
+                GeminiTool.from_dict(
+                    {
+                        "google_search_retrieval": google_search_config.dict(
```

Comment:
Why not simply `GoogleSearchConfig.parse_obj(static_function.configuration)`?

Reply:
`google_search_retrieval` is not related to `GoogleSearchConfig`, it's tool name